### PR TITLE
bugfix: patchListeners

### PR DESCRIPTION
**Pull request by wzshiming**
[X] Configuration Infrastructure
[X] Docs
[X] Installation
[X] Networking
[X] Performance and Scalability
[X] Policies and Telemetry
[X] Security
[X] Test and Release
[X] User Experience
[X] Developer Infrastructure

### CODE REVIEW

**Comment by wzshiming** [comment]
The result is not saved when editing the gateway, which will invalidate the EnvoyFilter applied to the gateway

**Comment by howardjohn** [comment]
@mandarjog haven't you been testing gateway EnvoyFilters though?

**Comment by mandarjog** [comment]
Yes I have been, is it that just chaining does not work ?
@wzshiming can you add a test case that brings out the error condition.
The fix seems correct, but we need a testcase anyway.

**Comment by mandarjog** [comment]
@wzshiming  Thanks for fixing it, Please add a test case.

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 1]
```diff
@@ -277,7 +277,7 @@ func (lb *ListenerBuilder) patchOneListener(l *listener.Listener, ctx networking
 }
 func (lb *ListenerBuilder) patchListeners() {
 	if lb.node.Type == model.Router {
-		envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, lb.node, lb.push, lb.gatewayListeners, false)
+		lb.gatewayListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, lb.node, lb.push, lb.gatewayListeners, false)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:280]: great, we can append listeners in `doListenerListOperation`. We should always set the return values.
Same applies to others (cluster, route), i have checked all others are ok

**Comment by wzshiming** [comment]
@hzxuzhonghu  @mandarjog PTAL

**Pull request merged**